lib/buildkite/test_collector/object.rb

```diff
@@ -7,7 +7,11 @@ def sleep(duration)
         tracer = Buildkite::TestCollector::Uploader.tracer
         tracer&.enter("sleep")
 
-        super
+        if duration
+          super(duration)
+        else
+          super()
+        end
```

Comment:
The new test isn't passing. Do we need to make the `duration` parameter optional? If we do that, can we simply call the `super` method without checking `duration`?

Reply:
Ooops I had the `duration = nil` locally, pushed a new change now.

Reply:
Is there anything we can do to simply this block and avoid nil checking `duration`?

Reply:
super will pass the nil (`duration` defaults to `nil` from method signature) down which will error 😞 

```
Traceback (most recent call last):
        5: from /Users/hhh/.gem/ruby/2.7.6/bin/irb:23:in `<main>'
        4: from /Users/hhh/.gem/ruby/2.7.6/bin/irb:23:in `load'
        3: from /Users/hhh/.rubies/ruby-2.7.6/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        2: from (irb):1
        1: from (irb):1:in `sleep'
TypeError (can't convert NilClass into time interval)
```